src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1340,6 +1342,7 @@ private Class<?> elementType() {
      * <li> its {@code ABSTRACT} and {@code FINAL} flags are present
      * <li> its {@code INTERFACE} flag is absent, even when the
      *      component type is an interface
+     * <li> its class file format version is that of the component class
```

Comment:
Please remove this requirement.

First, I'm not sure if it is true of the implementation. Even if it were true today, I don't think it is necessary to guarantee this as the class file format version is not directly retrievable by end-users (nor do I think it should be).

Reply:
For arrays, the implementation does use the cffv of the element type and it is a natural extension of the description of Class.accessFlags() using some of the modifiers (public, protected, and private) of the component type (as written a couple of lines above).  But it may be a bit of overreach to say that its appropriate for other modifiers of an array to have the same behavior.